### PR TITLE
Update faraday version in gemspec

### PR DESCRIPTION
**Pull request by d-m-u**
[Faraday got updated](https://github.com/lostisland/faraday/pull/1101) and [the manageiq folks](https://github.com/ManageIQ/manageiq/issues/19760) were hoping maybe that you all might be amenable to letting people update so we can support ruby 2.7.

Thanks!

### CODE REVIEW

**Comment by d-m-u** [comment]
WIP because it's failing tests.

**Comment by olleolleolle** [comment]
https://github.com/vcr/vcr/issues/788 

There are a few differences as well, which can be seen in tests (it wants `adapter` to be called, not `use`)

File: spec/lib/vcr/library_hooks/faraday_spec.rb [hunk 1]
```diff
@@ -43,8 +43,8 @@
 
   it 'prints a warning if the faraday connection stack contains a middleware after the HTTP adapter' do
     conn = Faraday.new(:url => 'http://sushi.com') do |builder|
-      builder.use Faraday::Adapter::NetHttp
-      builder.use Faraday::Response::Logger
+      builder.adapter Faraday::Adapter::NetHttp
+      builder.adapter Faraday::Response::Logger
```
Comment (olleolleolle) [inline spec/lib/vcr/library_hooks/faraday_spec.rb]: I believe this is a Middleware, whereas the line before has an Adapter.

Comment (d-m-u) [reply spec/lib/vcr/library_hooks/faraday_spec.rb]: Ah, thanks, let me try again 😆

**Comment by d-m-u** [comment]
Ah, I guess possibly https://github.com/vcr/vcr/blob/master/script/ci.sh#L89 needs to change?

**Comment by d-m-u** [comment]
Hey @olleolleolle thanks for the help, any chance you might be able to tell me what's up with [the last failure](https://travis-ci.org/vcr/vcr/jobs/646112809#L563) ?

**Comment by d-m-u** [comment]
https://github.com/lostisland/faraday/pull/750 changed the way that the handlers are stored. The default adapter is no longer stored in the list of handlers so the test that was previously failing is bogus, because it no longer warns, and these needed to be updated in order to no longer test that the adapter was in the list. Thanks @jrafanie for the 👀

**Comment by d-m-u** [comment]
@krainboltgreene could I get you to take a look at this, please?

File: script/ci.sh [hunk 2]
```diff
@@ -86,7 +86,7 @@ if [ "$RUBY_ENGINE" = "ruby" ]; then
     run script/test spec/lib/vcr/library_hooks/typhoeus_0.4_spec.rb
 fi
 
-BUNDLE_GEMFILE=Gemfile.faraday-0.11 fold "faraday-0.11" \
```
Comment (jrafanie) [inline script/ci.sh]: Is this old Gemfile.faraday-0.11 needed if we're adding the 1.0.0 version?

File: spec/lib/vcr/library_hooks/faraday_spec.rb [hunk 3]
```diff
@@ -10,45 +10,23 @@
...
-      VCR::Middleware::Faraday,
-      Faraday::Response::Logger,
-      Faraday::Adapter::NetHttp
-    ])
-  end
-
-  it 'prints a warning if the faraday connection stack contains a middleware after the HTTP adapter' do
-    conn = Faraday.new(:url => 'http://sushi.com') do |builder|
-      builder.use Faraday::Adapter::NetHttp
-      builder.use Faraday::Response::Logger
```
Comment (jrafanie) [inline spec/lib/vcr/library_hooks/faraday_spec.rb]: ✂️ Nice@!  Yeah, agreed.  This test is no longer valid with faraday 1.0.0 as the warning doesn't happen because the adapter is managed differently now.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
-require "codeclimate-test-reporter"
-CodeClimate::TestReporter.start
+require 'simplecov'
+SimpleCov.start
```
Comment (jrafanie) [inline spec/spec_helper.rb]: I'm late to this PR, was it failing on master because of this?

Comment (d-m-u) [reply spec/spec_helper.rb]: Yes. It was deprecated in https://github.com/codeclimate/ruby-test-reporter/blob/master/CHANGELOG.md#changes-1

**Comment by krainboltgreene** [comment]
@d-m-u Ready to merge?

**Comment by d-m-u** [comment]
@krainboltgreene should be good to go, yeah! Thanks!!

**Comment by d-m-u** [comment]
hey @krainboltgreene or @olleolleolle could we please get a version cut that uses the new faraday?

**Comment by jrafanie** [comment]
looks like v6.0.0 tag has it @d-m-u: https://github.com/vcr/vcr/blob/v6.0.0/CHANGELOG.md
